frontend/src/routes/index.tsx

```diff
@@ -1,40 +1,55 @@
-import { Navigate, createBrowserRouter } from 'react-router-dom';
+import { Navigate, createBrowserRouter, RouterProvider } from 'react-router-dom';
 import Login from '../pages/Login';
 import Dashboard from '../pages/Dashboard';
 import PublicOnlyRoutes from './PublicOnlyRoutes';
 import ProtectedRoutes from './ProtectedRoutes';
 import AuthRedirect from '../pages/AuthRedirect';
 import AppWrapper from './AppWrapper';
+import { pingProtectedBackend, pingPublicBackend } from '../api/questionBankApi';
+import { useAuth } from '../utils/hooks';
 
-export const router = createBrowserRouter([
-  {
-    element: <AppWrapper />,
-    children: [
-      {
-        element: <PublicOnlyRoutes />,
-        children: [
-          {
-            path: '/login',
-            element: <Login />,
-          },
-        ],
-      },
-      {
-        element: <ProtectedRoutes />,
-        children: [
-          {
-            path: '/dashboard',
-            element: <Dashboard />,
-          },
-        ],
-      },
-      { path: '*', element: <Navigate to="/login" replace /> },
-    ],
-  },
-  {
-    path: '/authRedirect',
-    element: <AuthRedirect />,
-  },
-]);
+// This is a wrapper that allows us to retrieve data from hooks that need to be passed into the router (e.g for use by data loaders). It violates
+// OCP because we have to modify this file every time we add a new hook that needs to be passed into the router.
+// However, it's a small price to pay for the convenience of being able to use hooks in the router.
+export default function PeerPrepRouterProvider() {
+  const authContext = useAuth();
+  const router = createBrowserRouter([
+    {
+      element: <AppWrapper />,
+      children: [
+        {
+          element: <PublicOnlyRoutes />,
+          children: [
+            {
+              path: '/login',
+              element: <Login />,
+            },
+          ],
+        },
+        {
+          element: <ProtectedRoutes />,
+          children: [
+            {
+              path: '/dashboard',
+              element: <Dashboard />,
+              loader: async () => {
+                // @TODO remove this data loader once we have a real data loader for the dashboard
+                return {
+                  protectedResponse: await pingProtectedBackend(authContext),
+                  publicResponse: await pingPublicBackend(authContext),
+                };
+              },
```

Comment:
Can we shift this loader function into `Dashboard.tsx`? so all the fetching logic and rendering logic are located in the same file albeit different functions

Reply:
Let's close this for now. I dont think our current choice of using React Contexts to store auth state allows us to use data loader functions, since we need the AuthContext to be rendered first before firing off the loader functions (loader functions are fired off in parallel).

To use loader functions, we probably need to use an external store such as redux